src/ui/TemplateMessageItemBody/index.tsx

```diff
@@ -0,0 +1,135 @@
+import './index.scss';
+import React, { ReactElement, useEffect, useState } from 'react';
+import type { BaseMessage } from '@sendbird/chat/message';
+import { getClassName } from '../../utils';
+import MessageTemplateWrapper from '../../modules/GroupChannel/components/MessageTemplateWrapper';
+import { MessageTemplateData, MessageTemplateItem } from './types';
+import restoreNumbersFromMessageTemplateObject from './utils/restoreNumbersFromMessageTemplateObject';
+import mapData from './utils/mapData';
+import selectColorVariablesByTheme from './utils/selectColorVariablesByTheme';
+import { SendbirdTheme } from '../../types';
+import useSendbirdStateContext from '../../hooks/useSendbirdStateContext';
+import { ProcessedMessageTemplate, WaitingTemplateKeyData } from '../../lib/dux/appInfo/initialState';
+import FallbackTemplateMessageItemBody from './FallbackTemplateMessageItemBody';
+import LoadingTemplateMessageItemBody from './LoadingTemplateMessageItemBody';
+
+const TEMPLATE_FETCH_RETRY_BUFFER_TIME_IN_MILLIES = 500; // It takes about 450ms for isError update
+
+interface TemplateMessageItemBodyProps {
+  className?: string | Array<string>;
+  message: BaseMessage;
+  isByMe?: boolean;
+  theme?: SendbirdTheme;
+}
+
+/**
+ * Returns copied message template object filled with given template data and color variables.
+ */
+const getFilledMessageTemplateWithData = (
+  template: MessageTemplateItem[],
+  templateData: Record<string, any>,
+  colorVariables: Record<string, string>,
+  theme: SendbirdTheme,
+): MessageTemplateItem[] => {
+  const selectedThemeColorVariables = selectColorVariablesByTheme({
+    colorVariables,
+    theme,
+  });
+  const parsedTemplate: MessageTemplateItem[] = mapData({
+    template: restoreNumbersFromMessageTemplateObject(template) as any,
+    source: { ...templateData, ...selectedThemeColorVariables },
+  }) as any;
+  return parsedTemplate;
+};
+
+export function TemplateMessageItemBody({
+  className = '',
+  message,
+  isByMe = false,
+  theme = 'light',
+}: TemplateMessageItemBodyProps): ReactElement {
+  const templateData: MessageTemplateData | undefined = message.extendedMessagePayload?.['template'] as MessageTemplateData;
+  if (!templateData?.key) {
+    return <FallbackTemplateMessageItemBody className={className} message={message} isByMe={isByMe} />;
+  }
+
+  const globalState = useSendbirdStateContext();
+  if (!globalState) {
+    return <FallbackTemplateMessageItemBody className={className} message={message} isByMe={isByMe} />;
+  }
+
+  const {
+    getCachedTemplate,
+    updateMessageTemplatesInfo,
+  } = globalState.utils;
+
+  const waitingTemplateKeysMap = globalState.stores.appInfoStore.waitingTemplateKeysMap;
+
+  const [
+    filledMessageTemplateItems,
+    setFilledMessageTemplateItems,
+  ] = useState<MessageTemplateItem[]>([]);
```

Comment:
This way, we can render it directly without drawing the loading screen.

```suggestion
  const [
    filledMessageTemplateItems,
    setFilledMessageTemplateItems,
  ] = useState<MessageTemplateItem[]>(() => {
    const cachedTemplate = getCachedTemplate(templateKey);
    if (cachedTemplate) {
      return getFilledMessageTemplateWithData(
          JSON.parse(cachedTemplate.uiTemplate),
          templateData.variables ?? {},
          cachedTemplate.colorVariables,
          theme,
        )
    } else {
      return [];
    }
  });
```

Reply:
good suggestion but loading screen is in the intended design
![Screenshot 2024-02-29 at 4 46 46 PM](https://github.com/sendbird/sendbird-uikit-react/assets/16806397/5dffdfa5-2512-405c-a89b-8c356caa6936)

Reply:
If data is available for rendering, we can render it immediately. We only recognize the need for loading when there's no template for a message received in real-time and we need to update

Reply:
oh yes i misread your code to call API but now i see it is not. Then this is good suggestion ill apply!